spec/core_ext/math/slope_spec.rb

```diff
@@ -0,0 +1,20 @@
+describe Math do
+  it ".slope_x_intercept" do
+    expect(described_class.slope_x_intercept(1, 1, 0)).to eq(1)
+    expect(described_class.slope_x_intercept(2, 1, 0)).to eq(2)
+    expect(described_class.slope_x_intercept(10, 1, 0)).to eq(10)
+    expect(described_class.slope_x_intercept(10, 2, 0)).to eq(5)
+  end
+
+  it ".slope_y_intercept" do
+    expect(described_class.slope_y_intercept(1, 1, 0)).to eq(1)
+    expect(described_class.slope_y_intercept(2, 1, 0)).to eq(2)
+    expect(described_class.slope_y_intercept(10, 1, 0)).to eq(10)
+    expect(described_class.slope_y_intercept(10, 2, 0)).to eq(20)
+  end
+
+  it ".slope" do
+    expect(described_class.slope([1.0, 2.0], [1.0, 2.0])).to eq([1.0, 0.0, 1.0])
+    expect(described_class.slope([1.0, 1.5], [1.0, 2.0])).to eq([2.0, -1.0, 1.0])
```

Comment:
ohhhh...this is not a slope, and I think that's what confused me.  This is actually a linear trendline calculation where r is the **r squared** value http://condor.depaul.edu/sjost/it223/documents/correlation.htm

Let's rename this method to `linear_trend`, then we can rename the params appropriately.